File: tasket/models/log.py
from dataclasses import dataclass
from enum import IntEnum
from typing import Optional

class LogSource(IntEnum):
    UNKNOWN = 0
    TASK = 1
    SERVER = 2

@dataclass
class Log:
    id: str
    task_id: str
    source: LogSource
    content: bytes
    created_at: int


@dataclass
class LogFilter:
    source: Optional[LogSource] = None
    contains: Optional[bytes] = None
    created_after: Optional[int] = None
    created_before: Optional[int] = None

File: src/model/log.go
package model

type LogSource uint

const (
	LogSource_UNKNOWN LogSource = 0
	LogSource_TASK    LogSource = 1
	LogSource_SERVER  LogSource = 2
)

type Log struct {
	Id        string    `json:"id"`
	TaskId    string    `json:"taskId"`
	Source    LogSource `json:"source"`
	Content   []byte    `json:"content"`
	CreatedAt int64     `json:"createdAt"`
}

type LogOptions struct {
	Source        *LogSource `json:"source"`
	CreatedAfter  *int64     `json:"createdAfter"`
	CreatedBefore *int64     `json:"createdBefore"`
}
